### PR TITLE
Rework debian packagemanager

### PR DESCRIPTION
**Pull request by loosebazooka**
- Use a non-bazel tool to generate bazel config to each deb
  individually. It will also be responsible for managing updates
  
  this is just the go part of #926

### CODE REVIEW

File: debian_package_manager/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# debian-package-manager
+
+A debian package index parser with bazel config generation
+
+# Usage
+
+```sh
+# if new versions are available, update snapshots and bazel config
+$ go build -o update cmd/update
```
Comment (priyawadhwa) [inline debian_package_manager/README.md:9]: should we just use `make update` here and have the make command run both of these?

File: debian_package_manager/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+$ ./update 
+
+# generate new config for current snapshot
+$ go build -o generate cmd/generate
```
Comment (priyawadhwa) [inline debian_package_manager/README.md:13]: similarly we could just have `make generate` here

File: debian_package_manager/internal/build/bazel_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/GoogleContainerTools/distroless/debian_package_manager/internal/deb"
+	"github.com/google/go-cmp/cmp"
+)
+
+func testdata() map[string]map[string]map[string]*deb.Package {
```
Comment (priyawadhwa) [inline debian_package_manager/internal/build/bazel_test.go:11]: `map[string]map[string]map[string]*deb.Package` is wild lol

**Review by priyawadhwa — APPROVED** [review]

File: debian_package_manager/README.md (hunk 1 above)
Comment (loosebazooka) [reply debian_package_manager/README.md:9]: oh yeah good call

File: debian_package_manager/internal/build/bazel_test.go (hunk 3 above)
Comment (loosebazooka) [reply debian_package_manager/internal/build/bazel_test.go:11]: This makes printing versions pretty easy (which may not be needed once we go full sbom? then we can just add a sorting function to deb.Package and we should get reproducible config generation).

**Pull request merged**